src/crypto/aes_gcm.go

```diff
@@ -4,22 +4,25 @@ import (
 	"crypto/aes"
 	"crypto/cipher"
 	"crypto/rand"
-	"crypto/sha256"
-	"crypto/sha512"
 	"encoding/json"
 	"fmt"
-	"hash"
 
+	"github.com/bndw/pick/crypto/pbkdf2"
+	"github.com/bndw/pick/crypto/scrypt"
 	"github.com/bndw/pick/errors"
-	"golang.org/x/crypto/pbkdf2"
 )
 
 type AESGCMClient struct {
-	settings AESGCMSettings
+	settings      AESGCMSettings
+	keyDerivation KeyDerivation
 }
 
 type AESGCMSettings struct {
-	KeyLen           int    `json:"keylen,omitempty" toml:"keylen"`
+	KeyLen        int            `json:"keylen,omitempty" toml:"keylen"`
+	KeyDerivation string         `json:"keyderivation,omitempty" toml:"keyderivation"`
+	PBKDF2        *pbkdf2.PBKDF2 `json:"pbkdf2,omitempty" toml:"pbkdf2"`
+	Scrypt        *scrypt.Scrypt `json:"scrypt,omitempty" toml:"scrypt"`
+	// These three Pbkdf2 configs are required for backwards-compatiblity :(
```

Comment:
Let's add a `deprecated` note here and we'll cut them off in the [near] future. I'm not extremely concerned with backward-compatibility at this stage, as this codebase is new and still under heavy churn.